### PR TITLE
[Driver] Add option -fsystem-debug

### PR DESCRIPTION
**Pull request by LU-JOHN**
Add option -fsystem-debug to complement -fno-system-debug.

### CODE REVIEW

File: clang/test/Driver/system_debug.c [hunk 1]
```diff
@@ -0,0 +1,12 @@
+// Ensure -no-system-debug is passed to cc1 appropriately
+
+// RUN: %clang -fsystem-debug    -### -c %s 2>&1 | FileCheck %s --check-prefix=CHECK-POS
```
Comment (srividya-sundaram) [inline clang/test/Driver/system_debug.c:3]: nit: extra spaces between `-fsystem-debug` and `-###`

Comment (LU-JOHN) [reply clang/test/Driver/system_debug.c]: Extra spaces removed.

**Review by srividya-sundaram — APPROVED** [review]

File: clang/include/clang/Driver/Options.td [hunk 2]
```diff
@@ -7172,6 +7174,9 @@ let Visibility = [CC1Option, CC1AsOption] in {
 def debug_info_macro : Flag<["-"], "debug-info-macro">,
   HelpText<"Emit macro debug information">,
   MarshallingInfoFlag<CodeGenOpts<"MacroDebugInfo">>;
+def no_system_debug : Flag<["-"], "no-system-debug">,
+  HelpText<"Do not emit debug information for declarations in system headers">,
+  MarshallingInfoFlag<CodeGenOpts<"NoSystemDebug">>;
```
Comment (mdtoguchi) [inline clang/include/clang/Driver/Options.td:7179]: Why add a new option here?  We should be able to use the old `-fsystem-debug` syntax for both command line and `-cc1` usage.

File: clang/include/clang/Driver/Options.td [hunk 3]
```diff
@@ -3844,10 +3844,12 @@ def fdebug_macro : Flag<["-"], "fdebug-macro">, Group<f_Group>,
 def fno_debug_macro : Flag<["-"], "fno-debug-macro">, Group<f_Group>,
   Visibility<[ClangOption, CLOption, DXCOption]>,
   HelpText<"Do not emit macro debug information">;
+def fsystem_debug : Flag<["-"], "fsystem-debug">, Group<f_Group>,
+  Visibility<[ClangOption, CC1Option, CLOption, DXCOption]>,
```
Comment (mdtoguchi) [inline clang/include/clang/Driver/Options.td:3848]:
```suggestion
  Visibility<[ClangOption, CLOption, DXCOption]>,
```
Not being passed to `-cc1`, no need for this to be visible at that level.

**Comment by bader** [comment]
@LU-JOHN, please, add [Driver] tag to the PR title.

File: clang/include/clang/Driver/Options.td (hunk 2 above)
Comment (LU-JOHN) [reply clang/include/clang/Driver/Options.td]: Removed -no-system-debug.

**Pull request merged**